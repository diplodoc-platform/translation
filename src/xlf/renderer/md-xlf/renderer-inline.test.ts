import {RenderParameters, render} from './index';

const baseRendererParameters = {
    source: {
        language: 'ru',
        locale: 'RU' as const,
    },
    target: {
        language: 'en',
        locale: 'US' as const,
    },
    markdownPath: 'text.md',
    skeletonPath: 'text.skl.md',
};

describe('renders xlf from markdown', () => {
    it('renders plain text', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер один. Предложение номер два.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders strong wrapped in <g> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер **один**. Предложение номер **два**.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders em wrapped in <g> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер *один*. Предложение номер *два*.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders s wrapped in <g> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер ~~один~~. Предложение номер ~~два~~.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders sup wrapped in <g> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер^один^. Предложение номер^два^.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders samp wrapped in <g> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер ##один##. Предложение номер ##два##.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders code wrapped in <x> tag', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер `один`. Предложение номер `два`.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders links wrapped in <g> and <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер [один](one.md "one"). Предложение номер [два](two.md).',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders ref links wrapped in <g> and <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер [{#T}](one.md "one"). Предложение номер [{#T}](two.md).',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders autolink wrapped in <g> and <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown:
                'Предложение номер один <https://www.google.com>. Предложение номер два <https://www.youtube.com>.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders variable href link wrapped in <g> and <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: 'Предложение номер [один]({{one}} "one"). Предложение номер [два]({{two}}).',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders image with href, title and size wrapped in <g> and <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown:
                'Sentence ![image](image.png "hint" =100x100). Sentence ![image](image.png "hint" =x100). Sentence ![image](image.png "hint" =100x). Sentence ![image](image.png =100x100). Sentence ![image](image.png "hint"). Sentence ![image](image.png). Sentence ![]().',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders video wrapped in <x> tag', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown:
                'Sentence goes here. Another sentence @[youtube](https://youtu.be/rJz4OaURJ6U)',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders heading with anchors wrapped in <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown: '# Heading with anchors {#anchor1} {#anchor2}',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders file wrapped in <g> and <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown:
                'Sentence with {% file src="path/to/file" name="readme.md" referrerpolicy="no-referrer" rel="noopener" target="_blank" type="text/plain" %}.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('renders liquid conditions wrapped in <x> tags', () => {
        const parameters: RenderParameters = {
            ...baseRendererParameters,
            markdown:
                'Sentence with {% if var == "val" %} val {% else %} other val {% endif %}. {% if var == "val" %} A {% else %} B {% endif %} Point.',
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });
});
