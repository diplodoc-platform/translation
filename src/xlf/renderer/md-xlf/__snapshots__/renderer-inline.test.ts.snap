// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders xlf from markdown renders code wrapped in <x> tag 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер <x ctype=\\"x-code\\" equiv-text=\\"\`один\`\\" />.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер <x ctype=\\"x-code\\" equiv-text=\\"\`два\`\\" />.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;

exports[`renders xlf from markdown renders em wrapped in <g> tags 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер <g ctype=\\"x-em\\" equiv-text=\\"*\\">один</g>.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер <g ctype=\\"x-em\\" equiv-text=\\"*\\">два</g>.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;

exports[`renders xlf from markdown renders links wrapped in <g> and <x> tags 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер <g ctype=\\"x-link_text_part\\" equiv-text=\\"[]\\">один</g><g ctype=\\"x-link_attributes_part\\" equiv-text=\\"()\\"><x ctype=\\"x-link_attributes_href\\" equiv-text=\\"&quot;one.md&quot;\\" /><g ctype=\\"x-link_attributes_title\\" equiv-text=\\"&quot;&quot;\\">one</g></g>.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер <g ctype=\\"x-link_text_part\\" equiv-text=\\"[]\\">два</g><g ctype=\\"x-link_attributes_part\\" equiv-text=\\"()\\"><x ctype=\\"x-link_attributes_href\\" equiv-text=\\"&quot;two.md&quot;\\" /></g>.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;

exports[`renders xlf from markdown renders plain text 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер один.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер два.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;

exports[`renders xlf from markdown renders s wrapped in <g> tags 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер <g ctype=\\"x-s\\" equiv-text=\\"~~\\">один</g>.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер <g ctype=\\"x-s\\" equiv-text=\\"~~\\">два</g>.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;

exports[`renders xlf from markdown renders samp wrapped in <g> tags 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер <g ctype=\\"x-samp\\" equiv-text=\\"##\\">один</g>.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер <g ctype=\\"x-samp\\" equiv-text=\\"##\\">два</g>.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;

exports[`renders xlf from markdown renders strong wrapped in <g> tags 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер <g ctype=\\"x-strong\\" equiv-text=\\"**\\">один</g>.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер <g ctype=\\"x-strong\\" equiv-text=\\"**\\">два</g>.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;

exports[`renders xlf from markdown renders sup wrapped in <g> tags 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"text.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"text.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Предложение номер<g ctype=\\"x-sup\\" equiv-text=\\"^\\">один</g>.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Предложение номер<g ctype=\\"x-sup\\" equiv-text=\\"^\\">два</g>.</source>
      </trans-unit>
    </body>
  </file>
</xliff>
"
`;
